components/apiserver-proxy/cmd/proxy/main.go

```diff
@@ -162,21 +146,32 @@ func main() {
 		glog.Fatalf("Failed to create authorizer: %v", err)
 	}
 
-	auth, err := proxy.New(kubeClient, cfg.auth, authorizer, authenticator)
+	authProxy, err := proxy.New(kubeClient, cfg.auth, authorizer, authenticator)
 
 	if err != nil {
 		glog.Fatalf("Failed to create rbac-proxy: %v", err)
 	}
 
-	proxy := httputil.NewSingleHostReverseProxy(upstreamURL)
+	proxyForApiserver := strings.Contains(cfg.upstream, proxy.KUBERNETES_SERVICE)
+
+	rp := httputil.NewSingleHostReverseProxy(upstreamURL)
+	
+	if proxyForApiserver {
```

Comment:
Are we going to use it in ant other way then as proxy for apiserver? If not I'd refrain from using conditions here and just hardcode this one case.

Reply:
yes, it should be possible to use it for another app as well although here it is designed for apisever :)